src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -27,8 +27,14 @@ public static partial class JsonSerializer
 
             if (state.Current.Enumerator == null)
             {
-                IEnumerable enumerable = (IEnumerable)jsonPropertyInfo.GetValueAsObject(state.Current.CurrentValue);
+                // Verify that the Dictionary can be serialized by having <string> as first generic argument.
```

Comment:
nit: move to a helper to avoid duplication.

Reply:
The only duplicated line I could find is `if (args.Length == 0 || args[0].UnderlyingSystemType != typeof(string))` - the line above and below are different. Doesn't seem worth it to create a helper method for one line.

Reply:
You are right. I thought the `Type[] args =...` and the exception were the same as well.